src/utility/util.jl

```diff
@@ -197,3 +157,26 @@ macro showtypeofgrad(x)
         end
     )
 end
+
+"""
+    @fwdthreads(ex)
+
+Apply `Threads.@threads` only in the forward pass of the program.
+
+It works by wrapping the for-loop expression in an if statement where in the forward pass
+the loop in computed in parallel using `Threads.@threads`, whereas in the backwards pass
+the `Threads.@threads` is omitted in order to make the expression differentiable.
+"""
+macro fwdthreads(ex)
+    @assert ex.head === :for "@fwdthreads expects a for loop:\n$ex"
+
+    diffable_ex = quote
+        if Zygote.isderiving()
+            $ex
+        else
+            Threads.@threads $ex
+        end
+    end
+
+    return esc(diffable_ex)
+end
```

Comment:
I like this idea, however I think we could do better by doing some more manual work. In particular, our use case involves a parallel map, where each element is independently computed and returned. Thus, the derivative is also just a parallel map, i.e. we could just compute the derivatives element-wise in parallel as well.

Nevertheless, that might require a bit more work, so I think it is fine to keep this for now and change that later. I would like to go to something like [OhMyThreads.jl](https://github.com/JuliaFolds2/OhMyThreads.jl), if necessary with some extra `rrule`, so I'll already link that here.

Reply:
I really like the idea of having a differentiable parallel map; that was actually what I hoped to find somewhere but haven't found yet. Writing the simple macro was just a quick solution that already speeds up CTMRG significantly for large unit cells or larger bond dimensions. So I'm also fine with keeping it like that for now, but I would also love to have a differentiable parallel map at some point.